### ISSUE TITLE
Should Prisma list fields be nullable?

### ISSUE THREAD

**Issue by joseaquino** [MichalLytek/typegraphql-prisma#42]
When having a Prisma schema like this:
```prisma
model Plan {
  zipCodes String[]
}
```
And then generate the corresponding types with `typegraphql-prisma@0.9.1` it generates the following model:
```ts
@TypeGraphQL.ObjectType({
  isAbstract: true,
  description: undefined,
  simpleResolvers: undefined,
})
export class Plan {
  @TypeGraphQL.Field(_type => [String], {
    nullable: true,
    description: undefined,
  })
  zipCodes?: string[] | null;
}
```
But since list field types cannot be set as optional in Prisma, shouldn't the resulting model be like this?
```ts
@TypeGraphQL.ObjectType({
  isAbstract: true,
  description: undefined,
  simpleResolvers: undefined,
})
export class Plan {
  @TypeGraphQL.Field(_type => [String], {
    nullable: false,
    description: undefined,
  })
  zipCodes!: string[];
}
```
This is a simple example to illustrate the situation we are experiencing with lists/arrays

**Using:**
- `@prisma/cli@2.12.1`
- `@prisma/client@2.12.1`
- `typegraphql-prisma@0.9.1`

**Comment by MichalLytek** [comment]
@joseaquino That's how it's described in DMMF:
```
"isList": true,
"isRequired": false,
```
`typegraphql-prisma` is a thin translation layer between the types from Prisma (DMMF) and the emitted TypeGraphQL classes.

However I agree that one-to-many relations (lists) could be non-nullable as if there's no data, it returns empty array, not null. Same goes to prisma client return type of `prisma.user.findFirst({}).posts()` 😕

**Comment by joseaquino** [comment]
Right, it doesn't make sense for empty lists to return null.

@MichalLytek could you point me in the right direction to where it would be best to post this question. Should I post it directly in the Prisma project or is there a specific repo for Prisma's DMMF descriptions?

Much appreciate your input 🙌🏼

**Comment by MichalLytek** [comment]
Related issue:
https://github.com/prisma/prisma/issues/4808

**Comment by MichalLytek** [comment]
Fixed by Prisma, released in `v0.10.2` 🚀
